src/core/MIGRATION.md

```diff
@@ -653,20 +671,66 @@ At the very least, any plugin exposing an extension point should do so with firs
 
 ### How is static code shared between plugins?
 
-Plugins are strongly discouraged from sharing static code for other plugins to import. There will be times when it is necessary, so it will remain possible, but it has serious drawbacks that won't necessarily be clear at development time.
+**tl;dr** Yes, but it should be limited to pure functional code that does not depend on outside state from the platform or a plugin.
```

Comment:
As I understand it is not only when plugin/platform state is involved, but even if platform exported a pure function, that pure function from platform cannot be used in statically exported plugin code.

Reply:
It could work technically, but we do not plan to export any static code from the platform itself. The platform's primary funcion is handling stateful services necessary to run Kibana apps, not providing utilities necessary to build Kibana apps. EUI is a great example of utility code that is not in the platform's purview but virtually all plugins depend on.

Reply:
It could work technically, but we do not plan to export any static code from the platform itself. The platform's primary funcion is handling stateful services necessary to run Kibana apps, not providing utilities necessary to build Kibana apps. EUI is a great example of utility code that is not in the platform's purview but virtually all plugins depend on.